jest.config.ts

```diff
@@ -37,16 +37,84 @@ function jestGithubRunnerSpecs(): JestConfig {
   };
 }
 
+interface ShardConfig {
```

Comment:
### Bug with `--shard` and `coverageThreshold` happens here maybe?

Did you check that the test coverage reports are still good, and that they check if everything is covered? As far as I understand there's a bug when you run Jest with `--shard` _and_ a `coverageThreshold` set:

-  https://github.com/facebook/jest/issues/12751

Maybe the same, or similar bug happens here?

Reply:
With our approach, we're basically apply *chroot* logic to the tests run.

The only problem with threshold I see is that cross-shard coverage doesn't count anymore, so we have to write *fair* tests, e.g. test `versioning` files inside `versioning` folder. This way, for example, if manager tests call versioning API, it won't improve versioning coverage anymore.

It's the side effect, but from code quality perspective, it actually looks good to me.
But it means we have to lower our threshold values per-shard and pay this technical debt eventually.

Reply:
@rarkins what do you think? Reducing the test threshold seems a big change to me. We should be careful that keep test coverage high, even if the coverage system can't enforce it automatically.

I like the idea of "fair tests". I'll let the maintainers chime in if this "fair test" thing is do-able.